app/controllers/concerns/api/articles_controller.rb

```diff
@@ -118,6 +123,21 @@ def unpublish
       end
     end
 
+    def search
```

Comment:
I left some comments here because I think that we can improve some code here and _possibly_ merge this with the index route. However, it would need some more work and we don't want to allow unnecessary params in for this iteration hence this seemed like the most sensible route.

Reply:
Unfortunately this is also going into the v0 and v1 api because ChatGPT seems to be not reading or overriding the Accept header that we specify for v1.